### ISSUE TITLE
Multiple related form inputs are not nested under a fieldset

### ISSUE THREAD

**Issue by alexadavidson** [cds-snc/platform-forms-client#5683]
**issue**:
When there are multiple form inputs that are related, they should be nested under a fieldset with a legend

**client feedback**:
> There is no fieldset component that could be used, the preset component titles for Contact Us, Address, Individual names are headings but the fields within are not grouped together in a fieldset, so that a screen reader would read the fieldset legend to indicate the fields are part of that section. This specifically helps with tabbing through the forms as the non interactive components does not get focus so the headings and any text outside of form elements can be skipped. We are using the form extensively and we find this is a great tool and wish the form makes use of HTML native tags for us to use. This helps with implementing default features that is available within, for example, using the min max attributes, inserting HTML tags like fieldset to setup relates fields.

**source**:
[freshdesk ticket 23035](https://cds-snc.freshdesk.com/a/tickets/23025)

**Comment by anikbrazeau** [comment]
Nice opportunity to add additional grouping structure and align with GCDS as well: https://design-system.alpha.canada.ca/en/components/fieldset/

We might consider implementing for the following form elements:
- Contact
- Individual names
- Address

Ones that are already treated as a set
- Verified address (AddressComplete)
- Date
- Repeating sets

**Comment by anikbrazeau** [comment]
Related #4218
